src/WebUI/Components/ServiceComponent.tsx

```diff
@@ -14,17 +14,44 @@ import * as Resources from "../Resources";
 import { IServiceItem, IVssComponentProperties } from "../Types";
 import { Utils } from "../Utils";
 import "./ServiceComponent.scss";
+import { V1PodList, V1Pod } from "@kubernetes/client-node";
+import { ListComponent } from "./ListComponent";
+import { IColumn } from "azure-devops-ui/Components/VssDetailsList/VssDetailsList.Props";
+import { Ago } from "azure-devops-ui/Ago";
+import { ColumnActionsMode } from "office-ui-fabric-react/lib/DetailsList";
+import { Statuses, StatusSize, Status } from "azure-devops-ui/Status";
+import { Tooltip } from "azure-devops-ui/TooltipEx";
 
 export interface IServiceComponentProperties extends IVssComponentProperties {
     service: IServiceItem;
+    podListingPromise: Promise<V1PodList>;
 }
 
-export class ServiceComponent extends BaseComponent<IServiceComponentProperties> {
+export interface IServiceComponentState {
+    showAssociatedPods:boolean;
+    pods:Array<V1Pod>;
+}
+
+const podNameKey: string = "svc-pod-name-key";
+const podImageKey: string = "svc-pod-image-key";
+const podStatusKey: string = "svc-pod-status-key";
+const podCreatedTimeKey:string = "svc-pod-age-key";
+
+export class ServiceComponent extends BaseComponent<IServiceComponentProperties, IServiceComponentState> {
+    constructor(props: IServiceComponentProperties) {
+        super(props, {});
+        this.state = {
+            showAssociatedPods:false,
+            pods: []
+        };
+    }
+
     public render(): JSX.Element {
         return (
             <div className="service-main-content">
                 {this._getMainHeading()}
                 {this._getServiceDetails()}
+                {!!this.state.showAssociatedPods && this._getAssociatedPods()}

```

Comment:
!!this.state.showAssociatedPods && --> we do not need this, as we should show column names when data is not available